src/main/java/uk/gov/companieshouse/api/strikeoffobjections/service/IObjectionService.java

```diff
@@ -16,7 +18,13 @@
 import javax.servlet.http.HttpServletResponse;
 
 public interface IObjectionService {
-    Objection createObjection(String requestId, String companyNumber, String ericUserId, String ericUserDetails);
+    Objection createObjection(String requestId,
+                              String companyNumber,
+                              ObjectionUserDetails userDetails);
+
+    ObjectionUserDetails buildUserDetails(String ericUserId,
```

Comment:
Personally I would place this in a different (maybe new) service as it doesn't really have anything to do with an Objection.

Reply:
This is part of the objection -I'm not sure we would be getting anything better by doing that.

Reply:
Please see my comment higher up. I don't think we need this new public service method.

Reply:
Unfortineately we do need as we have to test this and we have to pass the data in somehow.